runtime/src/main/java/org/corfudb/runtime/object/MultiVersionObject.java

```diff
@@ -327,6 +327,10 @@ private void syncStreamUnsafe(long timestamp) {
                         addressSpace.addAddress(globalAddress);
                         materializedUpTo = globalAddress;
                         resolvedUpTo = globalAddress;
+                    } catch (IllegalStateException e) {
+                        // The caller catches this TrimmedException and resets the object before retrying.
```

Comment:
Can you also log the number of retries in `getSnapshotProxy` :

`SnapshotProxy[{}] encountered trimmed addresses {} during sync to`

Reply:
Good idea. It is now logging that information as well.